command/agent/agent.go

```diff
@@ -440,6 +440,14 @@ func (a *Agent) consulConfig() *consul.Config {
 		base.AutopilotConfig.DisableUpgradeMigration = *a.config.Autopilot.DisableUpgradeMigration
 	}
 
+	// set the src address for outgoing rpc connections
+	// to RPCAdvertise with port 0 so that outgoing
+	// connections use a random port.
+	// todo(fs): can base.RPCAdvertise be nil ?
+	base.RPCSrcAddr = new(net.TCPAddr)
```

Comment:
I think this is just a convoluted way to say that the advertise address is either the rpc address or the overwritten one. I'll double check but I think we can drop that if statement. There should be one place where we configure the configuration and then it needs to be immutable.